### PR TITLE
[StackHCI] Add Check for ARC extension in Remote Support Cmdlets

### PR DESCRIPTION
**Pull request by hienbuihb**
## Description

Add a check to see if ARC extension is installed on a stamp. If it is, avoid downloading remote support packages from storage account and use the ones brought in through ARC extension instead.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by hienbuihb** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Comment by YanaXu** [comment]
Hi @hienbuihb ,

If you are from Microsoft, please join your github account to Azure org. If you are not, please reply to this comment. 

- Link your GitHub Account with Microsoft Account on this page: [Open Source Management Portal / Account link](https://repos.opensource.microsoft.com/link)
- Join the below organization on this page: [Open Source Management Portal / Organizations](https://repos.opensource.microsoft.com/orgs)
  - Azure

**Comment by hienbuihb** [comment]
> Hi @hienbuihb ,
> 
> If you are from Microsoft, please join your github account to Azure org. If you are not, please reply to this comment.
> 
> * Link your GitHub Account with Microsoft Account on this page: [Open Source Management Portal / Account link](https://repos.opensource.microsoft.com/link)
> * Join the below organization on this page: [Open Source Management Portal / Organizations](https://repos.opensource.microsoft.com/orgs)
>   
>   * Azure

File: src/StackHCI/custom/stackhci.ps1 [hunk 1]
```diff
@@ -5563,7 +5563,11 @@ function Install-AzStackHCIRemoteSupport{
     param()
 
     Setup-Logging -LogFilePrefix "AzStackHCIRemoteSupportInstall" -DebugEnabled ($DebugPreference -ne "SilentlyContinue")
-    if(Assert-IsObservabilityStackPresent){
+
+    $agentInstallType = (Get-ItemProperty -Path "HKLM:\SYSTEM\Software\Microsoft\AzureStack\Observability\RemoteSupport" -ErrorAction SilentlyContinue).InstallType
+    $observabilityStackPresent = Assert-IsObservabilityStackPresent
+
+    if($observabilityStackPresent -or ($agentInstallType -eq "ArcExtension")){
         Write-InfoLog("Install-AzStackHCIRemoteSupport is not available.")
```
Comment (swashtek) [inline src/StackHCI/custom/stackhci.ps1:5571]: Write-InfoLog("Install-AzStackHCIRemoteSupport is not available. ObsStackPresent: <True or False>, AgentInstallType: <"ArcExtension" or ""> ")

**Review by swashtek — APPROVED** [review]
Looks good to me.

**Comment by hienbuihb** [comment]
message for changelog: Modified AzStackHCI Remote Support cmdlets to check whether ARC extension was installed. If ARC extension was installed, avoid downloading RemoteSupport package from storage account and use module brought in through ARC extension instead.

**Review by YanaXu — APPROVED** [review]

**Pull request merged**